src/redis-benchmark.c

```diff
@@ -1182,8 +1182,8 @@ static int fetchClusterConfiguration() {
         }
         if (myself) {
             node = firstNode;
-            if (node->ip == NULL && ip != NULL) {
-                node->ip = ip;
+            if (ip != NULL && strcmp(node->ip, ip) != 0) {
+                node->ip = sdsnew(ip);
```

Comment:
@ShooterIT i'm not too familiar with this area, but noticed this PR was neglected and and i would like to promote, please help me with the review.

one thing that's odd to me is that in the past we used to enter this block only if `node->ip` is NULL, and now we proceed to do `strcmp` without even checking if if's NULL or not. so maybe we need another `||` here.

another thing i wanna note is that we need to aim for redis-benchmark to be able to work with both new and old redis servers, so it should still be compatible with the old behavior of CLUSTER NODES too.

Reply:
Thanks @oranagra the ip of firstNode is `config.hostip`, it never is NULL, redis-benchmark will core dump in the pas if it is NULL, because we double free this `node->ip`.
Old version  before https://github.com/redis/redis/commit/8fdc857a9f83e59b3062d051dc0155dd53c89ea7, we won't get myself ip, so we need to judge the ip parsed from CLUSTER NODE  is empty or not